schema.sql

```diff
@@ -62,6 +75,49 @@ create constraint trigger max_users
     after insert or update on passwd
     for each row execute procedure check_max_users();
 
+-- prevent users and groups sharing the same name
+create function check_invalid_name_for_passwd() returns trigger
+    language plpgsql as $$
+    begin
+        if (exists(select 1 from "group" where new.name = name)) then
+            raise check_violation using message = 'group name already exists: '||new.name;
+        end if;
+	return new;
+    end $$;
+create function check_invalid_name_for_group() returns trigger
+    language plpgsql as $$
+    begin
+        if (exists(select 1 from passwd where new.name = name)) then
+            raise check_violation using message = 'username already exists: '||new.name;
+        end if;
+	return new;
+    end $$;
+create constraint trigger check_name_exists_passwd
+    after insert or update on passwd
+    for each row
+    execute procedure check_invalid_name_for_passwd();
+create constraint trigger check_name_exists_group
+    after insert or update on "group"
+    for each row
+    execute procedure check_invalid_name_for_group();
+
+-- prevent users from taking names typically reserved for sysadmins
+CREATE TABLE "banned_usernames" (
+    "name" username_t UNIQUE NOT NULL
+);
+create function check_banned_username() returns trigger
+    language plpgsql as $$
+    begin
+        if (exists(select 1 from banned_usernames where name = new.name)) then
+            raise check_violation using message = 'username banned: '||new.name;
+        end if;
+        return new;
+    end $$;
+create constraint trigger check_banned_username
+    after insert on passwd
```

Comment:
You also need to check after an insert in `banned_usernames`

Reply:
Why? It already checks with the `UNIQUE` constraint whether or not it already exists.

Reply:
No, I meant that when inserting in `banned_usernames`, we should check that the username doesn't already exist.

Reply:
Oh, alright, will fix tomorrow hopefully?

Reply:
I just fixed it